disable hardcoded compiler/linker flags.  We handle stripping and debug
(-g), as well as optimizations (-On), by ourselves.

Remove architecture from the program names

Index: SConstruct
--- SConstruct.orig
+++ SConstruct
@@ -722,7 +722,7 @@ if env.msvc:
         env.Append(LINKFLAGS=["/OPT:REF"])
     elif env["optimize"] == "debug" or env["optimize"] == "none":
         env.Append(CCFLAGS=["/Od"])
-else:
+elif False: # disable -g/-O/-s flags; we handle them ourselves
     if env["debug_symbols"]:
         if env["platform"] == "windows":
             if methods.using_clang(env):
@@ -908,7 +908,6 @@ if env.dev_build:
 if env["precision"] == "double":
     suffix += ".double"
 
-suffix += "." + env["arch"]
 
 if not env["threads"]:
     suffix += ".nothreads"
